### PR TITLE
Adding support for i18n-debug

### PR DESCRIPTION
**Pull request by jeremyf**
See https://github.com/fphilipe/i18n-debug. We may need to observe if
this adds any significant time to builds.

Below is example output that is written to
`.internal_test_app/log/test.log`:

```console
[i18n-debug] en.simple_form.labels.file_set.files => nil
[i18n-debug] en.simple_form.labels.file_set.files => nil
[i18n-debug] en.simple_form.labels.defaults.files => "Upload a file"
[i18n-debug] en.simple_form.required.text => "required"
[i18n-debug] en.simple_form.required.mark => "*"
[i18n-debug] en.simple_form.required.html => "<span class=\"label label-info required-tag\">required</span>"
[i18n-debug] en.simple_form.hints.file_set.files => nil
[i18n-debug] en.simple_form.hints.file_set.files => nil
[i18n-debug] en.simple_form.hints.defaults.files => nil
[i18n-debug] en.simple_form.hints.defaults.files => nil
[i18n-debug] en.simple_form.hints.file_set.files => nil
[i18n-debug] en.simple_form.hints.file_set.files => nil
[i18n-debug] en.simple_form.hints.defaults.files => nil
[i18n-debug] en.simple_form.hints.defaults.files => nil
[i18n-debug] en.activemodel.models.file_set => nil
[i18n-debug] en.activemodel.models.active_fedora/base => nil
[i18n-debug] en.helpers.submit.file_set.update => nil
[i18n-debug] en.helpers.submit.update => "Save"
```

While the specs are running, I was running the following command to see
what translation keys was hit or missed:

```console
tail -f .internal_test_app/log/test.log
```

@projecthydra-labs/hyrax-code-reviewers

### CODE REVIEW

**Comment by jeremyf** [comment]
Related to https://github.com/projecthydra-labs/hyku/pull/629

**Comment by hackmastera** [comment]
@jeremyf how frequent is that output? is that the full output right now? It looks like it may have the potential to make the (already noisy) logs very noisy. Did you consider a report (rake task?). Or are you hoping the noise will cause us to address issues with i18n?

**Comment by jeremyf** [comment]
@hackmastera At present, anytime there is a request to I18n it would be logged.

It would be possible to require an ENV be set for the logging to occur (e.g. `require 'i18n-debug' if ENV['i18n_debug']`)

**Comment by hackmastera** [comment]
+1 on the ENV idea; I would even be okay with the ENV turning it off if more people want to see this in the logs on a regular basis (i.e. expect to incorporate whatever will be required to deal with the messages into their workflow)

**Comment by mjgiarlo** [comment]
@hackmastera @jeremyf Made the change y'all discussed. I know @jcoyne also discussed this with @jeremyf but I couldn't find that thread, so I'd be glad to hear your thoughts on this too.

Note that I also created this wiki page as a companion to this PR: https://github.com/projecthydra-labs/hyrax/wiki/Testing-internationalization-(i18n)-support

**Comment by mjgiarlo** [comment]
Rebased.

File: hyrax.gemspec [hunk 1]
```diff
@@ -80,4 +80,5 @@ Gem::Specification.new do |spec|
   spec.add_development_dependency 'shoulda-matchers', '~> 3.1'
   spec.add_development_dependency 'rails-controller-testing', '~> 0'
   spec.add_development_dependency 'webmock'
+  spec.add_development_dependency 'i18n-debug'
```
Comment (jcoyne) [inline hyrax.gemspec]: How about we don't even install it on travis? https://docs.travis-ci.com/user/languages/ruby/#Speeding-up-your-build-by-excluding-non-essential-dependencies

Comment (mjgiarlo) [inline hyrax.gemspec]: @jcoyne Would adding `unless ENV['TRAVIS'] == 'true'` at the end of this line suffice?

Comment (jcoyne) [inline hyrax.gemspec]: I think that would work.

Comment (mjgiarlo) [inline hyrax.gemspec]: Done

**Comment by mjgiarlo** [comment]
@jcoyne Re-review?

File: hyrax.gemspec [hunk 2]
```diff
@@ -79,4 +79,5 @@ Gem::Specification.new do |spec|
   spec.add_development_dependency 'shoulda-matchers', '~> 3.1'
   spec.add_development_dependency 'rails-controller-testing', '~> 0'
   spec.add_development_dependency 'webmock'
+  spec.add_development_dependency 'i18n-debug' unless ENV['TRAVIS'] == 'true'
```
Comment (jcoyne) [inline hyrax.gemspec]: why are we checking for a specific value here, and just the existence of the variable in `spec_helper.rb`?

Comment (mjgiarlo) [inline hyrax.gemspec]: Because `spec_helper.rb` was itself internally inconsistent. 😆  Fixed that in my follow-up. Thx.

**Comment by mjgiarlo** [comment]
@jcoyne OK, think I've got it now.

File: spec/spec_helper.rb [hunk 3]
```diff
@@ -51,6 +51,7 @@ def coverage_needed?
 require 'webmock/rspec'
 WebMock.disable_net_connect!(allow_localhost: true)
 
+require 'i18n/debug' if ENV['I18N_DEBUG']
```
Comment (mjgiarlo) [inline spec/spec_helper.rb]: Does this introduce a weirdness if we add `ENV['I18N_DEBUG']` in Travis? I.e., should the `hyrax.gemspec` change above also check the value of `ENV['I18N_DEBUG']`, or rectify this when needed?

Comment (jcoyne) [inline spec/spec_helper.rb]: NBD ATM

**Pull request merged**